<launch>
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="x_pos" default="0.5"/>
  <arg name="y_pos" default="3.5"/>
  <arg name="z_pos" default="0.0"/>
	
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="/home/gasser/catkin_ws/src/Milestone5/Mapnew/ID_world.world"/>
    <arg name="paused" value="false"/>
    <arg name="use_sim_time" value="true"/>
    <arg name="gui" value="true"/>
    <arg name="headless" value="false"/>
    <arg name="debug" value="false"/>
  </include>

  <param name="robot_description" command="$(find xacro)/xacro --inorder $(find turtlebot3_description)/urdf/turtlebot3_$(arg model).urdf.xacro" />

  <node pkg="gazebo_ros" type="spawn_model" name="spawn_urdf" args="-urdf -model turtlebot3_$(arg model) -x $(arg x_pos) -y $(arg y_pos) -z $(arg z_pos) -param robot_description" />
  <node pkg="Milestone5" name="BFS" type="BFS.py" output="screen">
    <param name="x_Goal" value="8.5" />   
    <param name="y_Goal" value="0.5"/>
   </node>

  <node pkg="Milestone5" name="Lyapunov" type="Lyapunov2.py">
    <param name="K_x" value="2" />   
    <param name="K_theta" value="2" />
    <param name="Vd_des" value="0.001" />   
    <param name="Omega_des" value="0" />
   </node>

</launch>
